nixos/modules/tasks/filesystems/zfs.nix

```
@@ -203,18 +204,26 @@ in
 
   imports = [
     (mkRemovedOptionModule [ "boot" "zfs" "enableLegacyCrypto" ] "The corresponding package was removed from nixpkgs.")
+    # Introduce me in 24.05.
+    # (mkRemovedOptionModule [ "boot" "zfs" "enableUnstable" ] "`boot.zfs.enableUnstable` was replaced by `boot.zfs.package` where you can specify `pkgs.zfsUnstable` to track the unstable ZFS.")
   ];
 
   ###### interface
 
   options = {
     boot.zfs = {
       package = mkOption {
-        readOnly = true;
         type = types.package;
-        default = if config.boot.zfs.enableUnstable then pkgs.zfsUnstable else pkgs.zfs;
-        defaultText = literalExpression "if config.boot.zfs.enableUnstable then pkgs.zfsUnstable else pkgs.zfs";
-        description = lib.mdDoc "Configured ZFS userland tools package.";
+        default = if cfgZfs.enableUnstable then pkgs.zfsUnstable else pkgs.zfs;
+        defaultText = literalExpression "if unstable is enabled then pkgs.zfsUnstable else pkgs.zfs";
```

```suggestion
        defaultText = literalExpression "if zfsUnstable is enabled then pkgs.zfsUnstable else pkgs.zfs";
```
otherwise people might confuse this with nixos-unstable